shapes/Circle.cpp

```diff
@@ -2,10 +2,6 @@
 #include <math.h>
 #include <iostream>
 
-Circle::Circle(double r)
-    : r_(r)
-{}
```

Comment:
You have removed this constructor, but you should use a delegation. I know that default argument version has the same behavior, but the idea here was to show constructor delegation feature.
You should leave this constructor and call below constructor on initializer list.
The same for other shapes.